### PR TITLE
Sweets/leaderboard

### PR DESCRIPTION
**Pull request by sweetmantech**

### CODE REVIEW

File: lib/alchemy/eth_getLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import axios from "axios"
+import getAlchemyBaseUrl from "./getAlchemyBaseUrl"
+
```
Comment (crazzywizard) [inline lib/alchemy/eth_getLogs.tsx:4]:
```suggestion
import { error } from "console"
```

File: lib/alchemy/eth_getLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  try {
+    const response = await axios.post(endpoint, payload)
+    const logs = response.data.result
+
+    return logs
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error("Error fetching logs:", error)
```
Comment (crazzywizard) [inline lib/alchemy/eth_getLogs.tsx]:
```suggestion
  } catch (err) {
    error("Error fetching logs:", err)
```

File: lib/getAffiliateMintEvents.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { BigNumber, utils } from "ethers"
+import { getAddress } from "ethers/lib/utils"
+import { ethGetLogs } from "./alchemy/eth_getLogs"
+
+export const getAffiliateMintEvents = async () => {
+  const eventSignature = utils.id("AffiliateSale(address,uint256,uint256,uint256)")
```
Comment (crazzywizard) [inline lib/getAffiliateMintEvents.tsx]: you change the emit right? This needs to be updated

**Review by crazzywizard — CHANGES_REQUESTED** [review]
Left some comments, I think one needs to be changed since the emit is changed.

File: lib/alchemy/eth_getLogs.tsx (hunk 1 above)
Comment (sweetmantech) [reply lib/alchemy/eth_getLogs.tsx:4]: updated ✅

File: lib/alchemy/eth_getLogs.tsx (hunk 2 above)
Comment (sweetmantech) [reply lib/alchemy/eth_getLogs.tsx]: updated ✅

File: lib/getAffiliateMintEvents.tsx (hunk 3 above)
Comment (sweetmantech) [reply lib/getAffiliateMintEvents.tsx]: updated ✅

**Review by crazzywizard — APPROVED** [review]

**Review by towardsnewtech — APPROVED** [review]

**Pull request merged**